mlibrary/search
# [LIBSEARCH-901] PRIORITY: Figure out what is going on with authentication and Get This

# Overview
> Multiple users, since July 26, have reported endless login links with Get This (in particular). Sometimes changing browsers resolves the problem, sometimes clearing cookies does. And sometimes neither approach works, with upwards of 10 passes through web authentication.

This pull request helps [LIBSEARCH-901](https://mlit.atlassian.net/browse/LIBSEARCH-901).

## `Login` Component
The `Login` component was never used. The link to use SSO was repeated in multiple areas. The component itself was renamed to `Authentication` and rebuilt so it can handle logging in, logging out, and styling.

## Class to Function
The components that use the new `Authentication` component were class components, and have been converted to functional components to align with [current React practices](https://react.dev/learn/your-first-component).

## Testing
- Make sure the PR is consistent in these browsers:
- Run accessibility tests:
- Get the site locally up and running:
  - Build the branch: `npm run build:dev`
  - Copy the build over to `spectrum`:
  ```
  cp ../search/search.local.tar.gz ./ 
  tar -xzf search.local.tar.gz -C public --transform=s%search/index.html%search/app.html% --strip-components=1 
  ```
  - Build `spectrum`: `docker-compose up --build --no-start`
  - Install `spectrum`: `docker-compose run --rm web bundle install`
  - Connect to VPN and run `spectrum`: `docker-compose up`
  - Go to [the site](http://localhost:3000/everything) and poke around:
    - Can you log in through the website header?
      - When you are logged in, does the link change to `Log out`?
      - When you click `Log out` and return to the site, does the link switch to `Log in`?
    - Can you log in through the `Actions` panel when clicking on `Email` or `Text`?
      - When you are logged in, can you view those panels?
    - Can you log in through Get This?
      - When you are logged in, can you view the request options?
